### PR TITLE
fix(CompositeDevice): Ensure proper handling of chord events in a way steam can understand them.

### PR DESCRIPTION
**Pull request by pastaq**
- Add default configuration for steam that covers 99% of handhelds.

### CODE REVIEW

**Review by ShadowApex — CHANGES_REQUESTED** [review]
The main issue I see with the current implementation is that the `sleep` calls that happen when performing these chords will halt the processing of all events for the CompositeDevice while its sleeping.

To avoid this, I propose we actually spawn a tokio task that will sleep for the desired amount of time, then send a `Command` using the sender side of the CompositeDevice channel to tell the device it should process the event after the sleep has finished.

e.g.

```rust
pub enum Command {
    ...
    WriteEvent(NativeEvent),
...
```

```rust
...
tokio::spawn(async move {
    tokio::time::sleep(Duration::from_millis(80)).await;
    if let Err(e) = tx.send(Command::WriteEvent(event)) {
        log::error!("Failed to send chord event command: {:?}", e);
    }
});
...
```

Then in the run loop, we can process the command sent through the channel after the delay:

```rust
match cmd {
    ...
    Command::WriteEvent(event) => {
        if let Err(e) = self.write_event(event).await {
            log::error!("Failed to write event: {:?}", e);
        }
    }
...
```

File: src/input/composite_device/mod.rs [hunk 1]
```diff
@@ -654,10 +656,12 @@ impl CompositeDevice {
         let events = if self.device_profile.is_some() {
             self.translate_event(&event).await?
         } else {
-            vec![event]
+            vec![event.clone()]
         };
+        // Check if we need to reverse the event list.
+        let events = self.process_event_list(&event, events);
```
Comment (ShadowApex) [inline src/input/composite_device/mod.rs:662]:
```suggestion
        // Translate the event using the device profile.
        let is_pressed = event.pressed();
        let mut events = if self.device_profile.is_some() {
            self.translate_event(&event).await?
        } else {
            vec![event]
        };

        // Check if we need to reverse the event list.
        if events.len() > 1 && is_pressed {
            events = events.into_iter().rev().collect();
        }
```

File: src/input/composite_device/mod.rs [hunk 2]
```diff
@@ -672,12 +676,57 @@ impl CompositeDevice {
...
+
+            // Add a keypress delay for synchronous events. This enables programming steam chords
+            // as steam will pass through events if not times correctly.
+            if events.len() > 1 {
+                //
+                let idx = events
+                    .iter()
+                    .position(|n| n.clone().as_capability() == event.as_capability());
+                if idx != Some(events.len() - 1) {
+                    sleep(Duration::from_millis(80));
```
[the remark this answers is not in the document]
Comment (pastaq) [reply src/input/composite_device/mod.rs]: I tried this, the tokio sleeps don't affect the thread we need to sleep so everything breaks.

**Review by ShadowApex — APPROVED** [review]
Ship it!

**Pull request merged**